dev_guide/app_tutorials/ruby_on_rails.adoc

```diff
@@ -0,0 +1,344 @@
+
+----
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: 5
+  host: localhost
+  username: rails
+  password:
+----
+
+Create your application's development and test databases by using this rake command:
+
+----
+$ rake db:create
+----
+
+This will create `development` and `test` database in your PostgreSQL server.
+
+=== Welcome page
+
+Since Rails 4 no longer serves a static `public/index.html` page in production, we need to create a 
+new root page.
+
+In order to have a custom welcome page we need to do following steps:
+
+* Create a *controller* with an index action 
+* Create a *view* page for the `welcome` controller `index` action
+* Create a *route* that will serve applications root page with the created *controller* and *view*  
+
+Rails offers a generator that will do all this necessary steps for you. Just type:
+
+----
+$ rails generate controller welcome index
+----
+
+All the necessary files have been created, now we just need to edit line 2 in `config/routes.rb`
+file to look like:
+----
+root 'welcome#index'
+---- 
+
+Run the rails server to verify the page is available
+
+----
+$ rails server
+----
+
+You should see your page by visiting link:http://localhost:3000[http://localhost:3000] in our browser.
+In case you don't see the page, check the logs that are output to your server to debug.
+
+=== Edit your App
+
+In order to have your application communicating with the PostgreSQL database service, you will need to edit the `default` section in your `config/database.yml` to use link:../../db_images/postgresql.html#environment-variables[environment variables], which you will define later, upon the database service creation.
+
+The `default` section in your edited `config/database.yml` together with pre-defined variables should look like:
+
+----
+<% user = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? "root" : ENV["POSTGRESQL_USER"] %>
+<% password = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? ENV["POSTGRESQL_ADMIN_PASSWORD"] : ENV["POSTGRESQL_PASSWORD"] %>
+<% db_service = ENV.fetch("DATABASE_SERVICE_NAME","").upcase %>
+
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  # For details on connection pooling, see rails configuration guide
+  # http://guides.rubyonrails.org/configuring.html#database-pooling
+  pool: <%= ENV["POSTGRESQL_MAX_CONNECTIONS"] || 5 %>
+  username: <%= user %>
+  password: <%= password %>
+  host: <%= ENV["#{db_service}_SERVICE_HOST"] %>
+  port: <%= ENV["#{db_service}_SERVICE_PORT"] %>
+  database: <%= ENV["POSTGRESQL_DATABASE"] %>
+----
+
+For example how the final file should look like, check out link:https://github.com/openshift/rails-ex[Ruby on Rails example application] link:https://github.com/openshift/rails-ex/blob/master/config/database.yml[config/database.yml].
+
+=== Store your App in Git
+
+OpenShift requires link:http://git-scm.com/[git], if you don't have it installed you will need to install it.
```

Comment:
This statement is too strong.  OpenShift does not require Git.  OpenShift requires application code be stored in git, so building applications from source requires git, indirectly.

Not sure how to rephrase though.  maybe:  "Building an application in OpenShift usually requires that the source code be stored in a link:http://git-scm.com/[git] repository, so you will need to install git if you do not already have it."